gemini/batch-prediction/intro_batch_prediction.ipynb

```diff
@@ -0,0 +1,588 @@
+        "```\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "qZlRIYsC01F1"
+      },
+      "source": [
+        "The example code below shows how to read all the .jsonl files in the Cloud Storage output location and print each JSON object."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "-jLl3es3dTqB"
+      },
+      "outputs": [],
+      "source": [
+        "# Get the bucket name\n",
+        "bucket_name = job.output_location.split(\"/\")[2]\n",
+        "\n",
+        "# Get the path after the bucket name\n",
+        "directory_path = \"/\".join(job.output_location.split(\"/\")[3:])\n",
+        "\n",
+        "storage_client = storage.Client()\n",
+        "bucket = storage_client.bucket(bucket_name)\n",
+        "\n",
+        "blobs = bucket.list_blobs(prefix=directory_path)\n",
+        "\n",
+        "for blob in blobs:\n",
+        "    if blob.name.endswith(\".jsonl\"):\n",
+        "        jsonl_string = blob.download_as_string().decode(\"utf-8\")\n",
+        "        for line in jsonl_string.splitlines():\n",
+        "            try:\n",
+        "                json_object = json.loads(line)\n",
+        "                print(json_object)\n",
+        "            except json.JSONDecodeError as e:\n",
+        "                print(e)"
+      ]
```

Comment:
This might be easier to load/use the data if it was loaded into a Pandas DataFrame. Something like this:

Install libraries at top of notebook:

```sh
pip install pandas gcsfs
```

```py
import gcsfs
import pandas as pd

# Specify the GCS directory
gcs_directory = job.output_location

# Use gcsfs to read all jsonl files directly into a single DataFrame
df = pd.concat(
    pd.read_json(f, lines=True, storage_options={"token": "cloud"})
    for f in gcsfs.GCSFileSystem().glob(gcs_directory + "*.jsonl")
)

# Display the first few rows of the final DataFrame
print(df.head())
```

Reply:
It's good to know that there is another way to do so! I will keep this in mind. Thanks.